### PR TITLE
Update EA Counting and Cron Code

### PR DESCRIPTION
**Pull request by bordoni**
_Ref:_ [C#66459](https://central.tri.be/issues/66459) [C#66457](https://central.tri.be/issues/66457)

### CODE REVIEW

File: src/Tribe/Aggregator/Cron.php [hunk 1]
```diff
@@ -306,16 +323,73 @@ public function verify_fetching_from_service() {
...
+	 * @see    http://wp-cli.org/docs/internal-api/#output
+	 *
+	 * @param  string $type    What kind of log is this
+	 * @param  string $message message displayed
+	 *
+	 * @return void
+	 */
+	public function log( $type = 'colorize', $message = '' ) {
+		// Only do something When in WP_CLI
+		if ( ! class_exists( 'WP_CLI' ) ) {
```
Comment (barryhughes) [inline src/Tribe/Aggregator/Cron.php]: If we're not in a WP_CLI environment, should we fall back on default logging? ie:

`Tribe__Main::instance()->log()->log_debug( 'Msg', 'src' );`

File: src/Tribe/Aggregator/Cron.php [hunk 2]
```diff
@@ -257,31 +257,47 @@ public function run() {
...
-		 * @todo Upon creation or modification of the schedule record, we need to store when should be the next
-		 */
+
 		$query = $records->query( array(
 			'post_status' => Tribe__Events__Aggregator__Records::$status->schedule,
 			'posts_per_page' => -1,
 		) );
 
 		if ( ! $query->have_posts() ) {
+			$this->log( 'debug', 'No Records Scheduled, skipped creating childs' );
```
Comment (barryhughes) [inline src/Tribe/Aggregator/Cron.php]: Should we allow translation of these strings?

File: src/Tribe/Aggregator/Errors.php [hunk 3]
```diff
@@ -56,5 +56,7 @@ private function __construct() {
 		tribe_register_error( 'core:aggregator:record-not-finalized', __( 'Import records must be finalized before posts can be inserted.', 'the-events-calendar' ) );
 		tribe_register_error( 'core:aggregator:save-child-failed', __( 'Unable to save scheduled import instance. Please try again.', 'the-events-calendar' ) );
 		tribe_register_error( 'core:aggregator:save-schedule-failed', __( 'Unable to save scheduled import. Please try again.', 'the-events-calendar' ) );
+
```
Comment (barryhughes) [inline src/Tribe/Aggregator/Errors.php]: No big deal, but why the extra line break - seems like the following line might belong in a block with the other tribe_register_error() calls?

File: src/Tribe/Aggregator/Record/Queue.php [hunk 4]
```diff
@@ -69,16 +69,28 @@ public function __construct( $record, $items = array() ) {
...
+		// Store the items retrieved
 		$this->items = $items;
 
 		// Count the Total of items now and stores as the total
 		$this->total = count( $this->items );
 	}
 
 	/**
+	 * Allows us to chech if the Events Data has still pending
+	 * @return boolean
```
Comment (barryhughes) [inline src/Tribe/Aggregator/Record/Queue.php]: Line break between the description and first tag.

**Comment by barryhughes** [comment]
Some optional changes, otherwise good to merge at your discretion :+1:

File: src/Tribe/Aggregator/Cron.php (hunk 2 above)
Comment (bordoni) [inline src/Tribe/Aggregator/Cron.php]: I don't think it's worth.

File: src/Tribe/Aggregator/Cron.php (hunk 1 above)
Comment (bordoni) [inline src/Tribe/Aggregator/Cron.php]: Will do that.

**Comment by barryhughes** [comment]
:football:

**Pull request merged**